src/server/db_slice.h

```diff
@@ -524,6 +525,39 @@ class DbSlice {
   void CallChangeCallbacks(DbIndex id, const ChangeReq& cr) const;
 
  private:
+  struct CounterCondFlag {
```

Comment:
I have some feedback about naming.
we already have `BlockingCounter` which has the same semantics. why not call it `LocalBlockingCounter` since it's just a single threaded  version of the same class.

Reply:
We currently use just a SharedMutex in the journal for the same problem. We can add a non-atomic mutex, switch to properly using lists or add this construct for both cases

Reply:
I will take care of this in separate PR. Nice @dranikpg

Reply:
https://github.com/dragonflydb/dragonfly/issues/3337 so I don't forget

Reply:
I actually think there is even more elegant solution to that that will void the need in any locking at all.
I think we can think of adding a quiescent notion to the transactional framework that will make sure calling callbacks during "safe" periods

Reply:
@romange happy to take this (but let's defer this until we get the other staff done). Can you plz expand your idea? I will create an issue to remember this :)

Reply:
Thought about it more. My approach won't work here because `RegisterOnChange` returns a version